src/arim/core.py

```diff
@@ -1344,19 +1356,21 @@ class BlockInImmersion(ExaminationObject):
     material : Material
         Alias for block_material
     couplant_material : Material
-    frontwall : OrientedPoints
-    backwall : OrientedPoints or None
+    walls : list[OrientedPoints]
```

Comment:
I see that in your case, the front and back walls are just two walls among many. But it could still be nice to index the walls by their name, for example `exam_obj.walls["Frontwall"]` instead of `exam_obj.walls[0]`.

Wouldn't it make more sense for `walls` to be a `dict[str, OrientedPoints]` instead of `list[OrientedPoints]`? It looks like `OrientedPoints.points.name` already acts as a de facto key, using a dict would make this more explicit (and prevent duplicated names).

Reply:
Very true. I had left it this way as it was used directly on the output from `make_contiguous_geometry()` but this does seem a more appropriate way to store the data, and would ensure walls have unique names by default.